### PR TITLE
Puppet master-slave test

### PR DESCRIPTION
**Pull request by dgutu-suse**
One step from https://progress.opensuse.org/issues/9464

### CODE REVIEW

File: products/sle/main.pm [hunk 1]
```diff
@@ -682,6 +682,9 @@ sub load_consoletests() {
         loadtest "console/zypper_ref";
         loadtest "console/yast2_lan" unless is_bridged_networking;
         loadtest "console/curl_https";
+        if (get_var('SCC_ADDONS', '') =~ /asmm/) {
```
Comment (okurz) [inline products/sle/main.pm]: you can actually use `check_var_array('SCC_ADDONS', 'asmm')`

File: tests/console/puppet.pm [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+echo 'server = puppetmaster.local' >> /etc/puppet/puppet.conf
+echo 'certname = puppetslave.local' >> /etc/puppet/puppet.conf
+echo '[master]' >> /etc/puppet/puppet.conf
+echo 'certname = puppetmaster.local' >> /etc/puppet/puppet.conf
+echo 'dns_alt_names = puppetmaster.local' >> /etc/puppet/puppet.conf
+systemctl start puppetmaster
+systemctl start puppet
+EOF
+    assert_script_run($_) foreach (split /\n/, $puppet_conf);
+    validate_script_output $output, sub { m/puppetslave\.local/ }, 3;
```
Comment (okurz) [inline tests/console/puppet.pm]: is there a specific reason for the "3"? I suggest to keep the default so at best just delete the variable.

File: tests/console/puppet.pm [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+echo '127.0.0.2 puppetmaster.local puppetslave.local' >> /etc/hosts
+echo 'server = puppetmaster.local' >> /etc/puppet/puppet.conf
+echo 'certname = puppetslave.local' >> /etc/puppet/puppet.conf
+echo '[master]' >> /etc/puppet/puppet.conf
+echo 'certname = puppetmaster.local' >> /etc/puppet/puppet.conf
+echo 'dns_alt_names = puppetmaster.local' >> /etc/puppet/puppet.conf
+systemctl start puppetmaster
+systemctl start puppet
+EOF
+    assert_script_run($_) foreach (split /\n/, $puppet_conf);
```
Comment (asmorodskyi) [inline tests/console/puppet.pm]: cool ! I really like this idea. Not in terms of this PR but generally we need to think about assert_script_run_multiline method or simply modify assert_script_run to detect multi line string and use this pattern in this case !

File: products/sle/main.pm (hunk 1 above)
Comment (dgutu-suse) [inline products/sle/main.pm]: Your proposal doesn't work, we have all modules not only asmm, so I'll stick with my original version.

**Comment by dgutu-suse** [comment]
Proof that its working :) 
http://markeb.arch.suse.de/tests/725

File: products/sle/main.pm (hunk 1 above)
Comment (asmorodskyi) [inline products/sle/main.pm]: did you actually try it ? probably you not correctly understand how it works - 
for example string 'hpcm asmm extratext' will return true for `check_var_array('SCC_ADDONS', 'asmm')`

File: tests/console/puppet.pm [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+# #
+# # Copyright © 2017 SUSE LLC
+# #
+# # Copying and distribution of this file, with or without modification,
+# # are permitted in any medium without royalty provided the copyright
+# # notice and this notice are preserved.  This file is offered as-is,
+# # without any warranty.
+#
+# # Summary: Test installation of puppet master and slave on the same host
+# # Maintainer: Dumitru Gutu <dgutu@suse.de>
```
Comment (mnowaksuse) [inline tests/console/puppet.pm]: Isn't the copyright notice commented-out a bit too much :)?

Comment (okurz) [inline tests/console/puppet.pm]: yes, should be only one '# '

File: tests/console/puppet.pm (hunk 3 above)
Comment (okurz) [inline tests/console/puppet.pm]: yeah but we should not to go overboard with that and rather use scripts in data/ and call that from there. Not necessary for this PR now.

**Pull request merged**